File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.cpp
#include <string>
#include <vector>

#include <algorithm> // reverse 함수 사용

using namespace std;

string solution(string my_string, int s, int e) {
    // s 인덱스부터 e 인덱스까지 (e + 1 지점 직전까지) 구간 뒤집기
    reverse(my_string.begin() + s, my_string.begin() + e + 1);
    
    return my_string;
}

File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.py
def solution(my_string, s, e):
    # s 전까지 + (s부터 e까지 뒤집은 것) + e 이후부터
    return my_string[:s] + my_string[s : e + 1][::-1] + my_string[e + 1 :]